frame/support/src/storage/mod.rs

```diff
@@ -427,6 +460,36 @@ pub trait IterableStorageNMap<K: ReversibleKeyGenerator, V: FullCodec>: StorageN
 	/// add elements to the map while doing this, you'll get undefined results.
 	fn drain() -> Self::Iterator;
 
+	/// The decode function used while iterating through the prefix key-value pairs in
+	/// [`IterableStorageNMap::iter_prefix`].
+	///
+	/// A type parameter `KP` denoting the type of the key prefix must be provided. See 
+	fn partial_key_value_decode_fn<KP>(
+		raw_key_without_prefix: &[u8],
+		raw_value: &[u8],
+	) -> Result<(<K as HasKeyPrefix<KP>>::Suffix, V), codec::Error>
+	where K: HasReversibleKeyPrefix<KP>;
+
+	/// The decode function used while iterating through the prefix keys in
+	/// [`IterableStorageNMap::iter_key_prefix`].
+	///
+	/// A type parameter denoting the type of the key prefix must be provided. See
```

Comment:
This documentation is not great, I'll need to rephrase this to properly explain what `KP` does.